codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/ModelMethodOverrides.java

```diff
@@ -81,14 +83,28 @@ public MethodSpec toStringMethod(ShapeModel shapeModel) {
                                                                            StringBuilder.class)
                                                              .addStatement("sb.append(\"{\")");
 
-        shapeModel.getNonStreamingMembers()
-                  .forEach(m -> {
-                      String getterName = m.getFluentGetterMethodName();
-                      toStringMethodBuilder.beginControlFlow("if ($N() != null)", getterName)
-                                           .addStatement("sb.append(\"$N: \").append($N()).append(\",\")", m.getName(),
-                                                         getterName)
-                                           .endControlFlow();
-                  });
+
+        if (!shapeModel.getNonStreamingMembers().isEmpty()) {
+            int size = shapeModel.getNonStreamingMembers().size();
+
+            shapeModel.getNonStreamingMembers()
+                    .subList(0, size - 1)
+                    .forEach(m -> {
+                        String getterName = m.getFluentGetterMethodName();
+                        toStringMethodBuilder.beginControlFlow("if ($N() != null)", getterName)
+                                .addStatement("sb.append(\"$N: \").append($N()).append(\",\")", m.getName(),
+                                        getterName)
+                                .endControlFlow();
+                    });
+
+
+            // Do not append comma at end of statement for last member
```

Comment:
Not sure this is going to work - what happens if both the last and second last members are null?

Reply:
Good callout. This will have extra comma. This behavior is same in v1 too.